File: my-folder/1544-count-good-nodes-in-binary-tree/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int goodNodes(TreeNode root) {
        return dfs(root, root.val);
    }

    private int dfs (TreeNode root, int max) {
        if (root == null) return 0;
        int res = 0;
        if (root.val >= max ) {
            res = 1;
            max = root.val;
        }

        res += dfs(root.left, max);
        res += dfs(root.right, max);

        return res;
    }
}

File: my-folder/1544-count-good-nodes-in-binary-tree/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        def dfs(node, max_val):
            if not node:
                return 0
            if node.val >= max_val:
                return 1 + dfs(node.left, node.val) + dfs(node.right, node.val)
            return dfs(node.left, max_val) + dfs(node.right, max_val)
        return dfs(root, root.val)
